### ISSUE TITLE
[Data] Improve the object store usage info 

### ISSUE THREAD

**Issue by scottsun94** [ray-project/ray#39804]
### Description

Slack thread: https://ray-distributed.slack.com/archives/CN2RGCHRR/p1695225640985159

`3.75 GB / 1.65 GB Object_store_memory` is confusing. We should make it clear that it includes the spilled data. Something like:

`3.75 GB / 1.65 GB Object Store Memory (2.10 GB spilled to external storage)`

**Comment by N3XT14** [comment]
Would love to take this issue. It would also be great if you could point me in some direction. 

Thanks

**Comment by scottsun94** [comment]
Thanks for the interest! @scottjlee for code pointer.

**Comment by scottjlee** [comment]
Hi @N3XT14 , thanks for your interest. The main code path involved is in `_report_current_usage` in [streaming_executor.py](https://github.com/ray-project/ray/blob/master/python/ray/data/_internal/execution/streaming_executor.py#L300). We can update `resources_status` here accordingly, and update all related unit tests which check this text.

You can take a look at [this page](https://docs.ray.io/en/latest/ray-contribute/getting-involved.html) for best practices contributing to Ray. Thanks again for your contributions!

**Comment by N3XT14** [comment]
Thanks @scottjlee. So the change is to just make the output more descriptive by providing the necessary information regarding the portion of the memory.

I do have certain questions about the information to be provided in the output.
Eg: 3.75 GB / 1.65 GB Object Store Memory (2.10 GB spilled to external storage)

so 2.10 which is the difference will always be "spilled to external storage" or should I look for where the remaining storage is assigned/present?

**Comment by scottjlee** [comment]
@N3XT14 yeah that's right, you can directly use the difference between `cur_usage.overall.object_store_memory_str()` and `limits.object_store_memory_str()`, which will be the excess spilled to external storage.
